packages/graphql/src/classes/Subgraph.ts

```diff
@@ -0,0 +1,291 @@
+
+        this.linkExtension = extension;
+
+        this.parseLinkImportArgument(linkDirective);
+    }
+
+    public getFullyQualifiedDirectiveName(
+        name: FederationDirectiveName
+    ): FederationDirectiveName | FullyQualifiedFederationDirectiveName | string {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return this.importArgument.get(name)!;
+    }
+
+    // TODO: Remove this function, parse `@shareable` directives and add them only on related generated types and fields
+    public augmentGeneratedSchemaDefinition(typeDefs: DocumentNode): DocumentNode {
+        const definitions: DefinitionNode[] = [];
+
+        const shareable = makeDirectiveNode(this.importArgument.get("shareable") as string, {}) as ConstDirectiveNode;
+
+        // This is a really filthy hack to apply @shareable
+        for (const definition of typeDefs.definitions) {
+            if (definition.kind === Kind.OBJECT_TYPE_DEFINITION) {
+                if (
+                    ["Query", "Mutation"].includes(definition.name.value) ||
+                    definition.name.value.endsWith("Connection") ||
+                    definition.name.value.endsWith("AggregateSelection") ||
+                    definition.name.value.endsWith("Edge")
+                ) {
+                    if (definition.directives) {
+                        definitions.push({
+                            ...definition,
+                            directives: [...definition.directives, shareable],
+                        });
+                    } else {
+                        definitions.push({
+                            ...definition,
+                            directives: [shareable],
+                        });
+                    }
+                } else {
+                    definitions.push(definition);
+                }
+            } else {
+                definitions.push(definition);
+            }
+        }
+
+        return {
+            ...typeDefs,
+            definitions,
+        };
+    }
+
+    public buildSchema({ typeDefs, resolvers }: { typeDefs: DocumentNode; resolvers: Record<string, any> }) {
+        return buildSubgraphSchema({
+            typeDefs,
+            resolvers,
+        });
+    }
+
+    public getReferenceResolvers(nodes: Node[], driver: neo4j.Driver): IResolvers {
+        const resolverMap: IResolvers = {};
+
+        const document = mergeTypeDefs(this.typeDefs);
+
+        document.definitions.forEach((def) => {
+            if (def.kind === Kind.OBJECT_TYPE_DEFINITION) {
+                resolverMap[def.name.value] = {
+                    __resolveReference: this.getReferenceResolver(nodes, driver),
+                };
+            }
+        });
+
+        return resolverMap;
+    }
+
+    private getReferenceResolver(
+        nodes: Node[],
+        driver: neo4j.Driver
+    ): (reference, context, info) => Promise<unknown | undefined> {
```

Comment:
The result type of this is going to be `Promise<unknown>` as `undefined` is a subset of `unknown`